lib/plugin/opsgenie/client.go

```diff
@@ -0,0 +1,281 @@
+/*
+Copyright 2015-2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package opsgenie
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/gravitational/teleport/integrations/lib"
+
+	"github.com/aws/aws-sdk-go/aws/defaults"
+	"github.com/go-resty/resty/v2"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/trace"
+)
+
+const (
+	alertKeyPrefix = "teleport-access-request"
+)
+
+var alertBodyTemplate = template.Must(template.New("alert body").Parse(
+	`{{.User}} requested permissions for roles {{range $index, $element := .Roles}}{{if $index}}, {{end}}{{ . }}{{end}} on Teleport at {{.Created.Format .TimeFormat}}.
+{{if .RequestReason}}Reason: {{.RequestReason}}{{end}}
+{{if .RequestLink}}To approve or deny the request, proceed to {{.RequestLink}}{{end}}
+`,
+))
+var reviewNoteTemplate = template.Must(template.New("review note").Parse(
+	`{{.Author}} reviewed the request at {{.Created.Format .TimeFormat}}.
+Resolution: {{.ProposedState}}.
+{{if .Reason}}Reason: {{.Reason}}.{{end}}`,
+))
+var resolutionNoteTemplate = template.Must(template.New("resolution note").Parse(
+	`Access request has been {{.Resolution}}
+{{if .ResolveReason}}Reason: {{.ResolveReason}}{{end}}`,
+))
+
+// Client is a wrapper around resty.Client.
+type Client struct {
+	ClientConfig
+
+	client *resty.Client
+}
+
+type ClientConfig struct {
+	// APIKey is the API key for Opsgenie
+	APIKey string
+	// APIEndpoint is the endpoint for the Opsgenie API
+	APIEndpoint string
```

Comment:
Can you provide some example or be more explicit about the expected format (are leading `/` acceptable? is `https://` required?)